sky/serve/serve_state.py

```diff
@@ -603,154 +727,182 @@ def get_replicas_at_status(
-    with db_utils.safe_cursor(_DB_PATH) as cursor:
-        rows = cursor.execute(
-            """\
-            SELECT spec FROM version_specs
-            WHERE service_name=(?)
-            AND version=(?)""", (service_name, version)).fetchall()
-    for row in rows:
-        return pickle.loads(row[0])
-    return None
+    assert _SQLALCHEMY_ENGINE is not None
+    with orm.Session(_SQLALCHEMY_ENGINE) as session:
+        result = session.execute(
+            sqlalchemy.select(version_specs_table.c.spec).where(
+                sqlalchemy.and_(
+                    version_specs_table.c.service_name == service_name,
+                    version_specs_table.c.version == version))).fetchone()
+    return pickle.loads(result[0]) if result else None
 
 
 @init_db
 def delete_version(service_name: str, version: int) -> None:
     """Deletes a version from the database."""
-    assert _DB_PATH is not None
-    with db_utils.safe_cursor(_DB_PATH) as cursor:
-        cursor.execute(
-            """\
-            DELETE FROM version_specs
-            WHERE service_name=(?)
-            AND version=(?)""", (service_name, version))
+    assert _SQLALCHEMY_ENGINE is not None
+    with orm.Session(_SQLALCHEMY_ENGINE) as session:
+        session.execute(
+            sqlalchemy.delete(version_specs_table).where(
+                sqlalchemy.and_(
+                    version_specs_table.c.service_name == service_name,
+                    version_specs_table.c.version == version)))
+        session.commit()
 
 
 @init_db
 def delete_all_versions(service_name: str) -> None:
     """Deletes all versions from the database."""
-    assert _DB_PATH is not None
-    with db_utils.safe_cursor(_DB_PATH) as cursor:
-        cursor.execute(
-            """\
-            DELETE FROM version_specs
-            WHERE service_name=(?)""", (service_name,))
+    assert _SQLALCHEMY_ENGINE is not None
+    with orm.Session(_SQLALCHEMY_ENGINE) as session:
+        session.execute(
+            sqlalchemy.delete(version_specs_table).where(
+                version_specs_table.c.service_name == service_name))
+        session.commit()
 
 
 @init_db
 def get_latest_version(service_name: str) -> Optional[int]:
-    assert _DB_PATH is not None
-    with db_utils.safe_cursor(_DB_PATH) as cursor:
-        rows = cursor.execute(
-            """\
-            SELECT MAX(version) FROM version_specs
-            WHERE service_name=(?)""", (service_name,)).fetchall()
-    if not rows or rows[0][0] is None:
-        return None
-    return rows[0][0]
+    assert _SQLALCHEMY_ENGINE is not None
+    with orm.Session(_SQLALCHEMY_ENGINE) as session:
+        result = session.execute(
+            sqlalchemy.select(sqlalchemy.func.max(
+                version_specs_table.c.version)).where(
+                    version_specs_table.c.service_name ==
+                    service_name)).fetchone()
+    return result[0] if result and result[0] is not None else None
```

Comment:
Could this be simplified to
```suggestion
    return result[0] if result and else None
```
because if `result[0]` is None then it returns None